src/server.ts

```diff
@@ -97,8 +137,28 @@ async function respondToQuestion(
 	const response = await openai.chat.completions.create({
 		model: 'gpt-4',
 		messages,
+		tools,
+		tool_choice: 'auto',
 	});
-	const answer = `${response.choices[0].message.content || ''}\n\nTo learn more, read:\n${urls}`;
+	let answer = response.choices[0].message.content || '';
+
+
+	if (response.choices[0].message.tool_calls) {
+		for (const toolCall of response.choices[0].message.tool_calls) {
+			if (toolCall.function.name === 'get_api_status') {
+				const status = await getApiStatus();
+				if (status) {
+					answer += `\n\nDiscord API Status:\n` +
```

Comment:
This is a great start!  Can we juice it a little and use an embed instead of plain text?  That can come after this PR lands if you like.